src/gluonnlp/vocab/bert.py

```diff
@@ -201,3 +204,70 @@ def from_json(cls, json_str):
         bert_vocab._cls_token = vocab_dict.get('cls_token')
 
         return bert_vocab
+
+    @classmethod
+    def from_sentencepiece(cls,
+                           path,
+                           mask_token=MASK_TOKEN,
+                           sep_token=SEP_TOKEN,
+                           cls_token=CLS_TOKEN,
+                           unknown_token=UNKNOWN_TOKEN,
+                           padding_token=PADDING_TOKEN,
+                           bos_token=None,
+                           eos_token=None,
+                           reserved_tokens=None):
+        """BERTVocab from pre-trained sentencepiece Tokenizer
+
+        Parameters
+        ----------
+        path : str
+            Path to the pre-trained subword tokenization model.
+        mask_token : hashable object or None, default '[MASK]'
+            The representation for the special token of mask token for BERT
+        sep_token : hashable object or None, default '[SEP]'
+            a token used to separate sentence pairs for BERT.
+        cls_token : hashable object or None, default '[CLS]'
+        unknown_token : hashable object or None, default '[UNK]'
+            The representation for any unknown token. In other words,
+            any unknown token will be indexed as the same representation.
+        padding_token : hashable object or None, default '[PAD]'
+            The representation for padding token.
+        reserved_tokens : list of strs or None, optional
+            A list of reserved tokens that will always be indexed.
+
+        Returns
+        -------
+        BERTVocab
+        """
+        sp = SentencepieceTokenizer(os.path.expanduser(path))
+        token_to_idx = {
+            convert_to_unicode(t): i
+            for i, t in enumerate(sp.tokens)
+        }
+        special_tokens_dic = {
+            k: convert_to_unicode(token) if token is not None else None
+            for k, token in zip(['mask', 'sep', 'cls', 'unk', 'pad', 'bos', 'eos'], [
```

Comment:
What i meant is that, the sentence piece vocab already added unk, bos, eos in its dictionary, right? You can get them by sth like
```
sp = sentencepiece()
sp.load('model')
bos_id, eos_id, unk_id =  sp.bosId(), sp.eosId(), sp.unkId()
bos_token, eos_token, unk_token = ... sp.tokens[bos_id], sp.tokens[eos_id], sp.tokens[unk_id]
```
And we should pass these bos_token, eos_token, unk_token from the sentence piece model to construct our vocab? This way the vocab is more consistent with the model from sentencepiece. Otherwise, there're '<s>' and '</s>' from sentencepiece vocab which already stand for bos, eos.

Reply:
Thanks. Your right.